apps/contact-form/middleware.ts

```diff
@@ -0,0 +1,38 @@
+
+import { NextRequest, NextResponse } from 'next/server';
+import { i18n, getLocale } from '@fdk-frontend/dictionaries';
+
+export const middleware = (request: NextRequest) => {
+  const pathname = request.nextUrl.pathname;
+
+  // // `/_next/` and `/api/` are ignored by the watcher, but we need to ignore files in `public` manually.
+  // // If you have one
+  // if (
+  //   [
+  //     '/manifest.json',
+  //     '/favicon.ico',
+  //     // Your other files in `public`
+  //   ].includes(pathname)
+  // )
+  //   return
+
+  // Check if there is any supported locale in the pathname
+  const pathnameIsMissingLocale = i18n.locales.every(
+    (locale: unknown) => (!pathname.startsWith(`/${locale}/`) && pathname !== `/${locale}`),
+  );
+
+  // Redirect if there is no locale
+  if (pathnameIsMissingLocale) {
+    const locale = getLocale(request);
+
+    // e.g. incoming request is /products
+    // The new URL is now /en-US/products
+    return NextResponse.redirect(new URL(`/${locale}${pathname.startsWith('/') ? '' : '/'}${pathname}`, request.url));
+  }
+  return null;
+}
+
+export const config = {
+  // Matcher ignoring `/_next/` and `/api/`
+  matcher: ['/((?!api|_next/static|_next/image|favicon.ico).*)'],
+};
```

Comment:
Trenger vi denne middlewaren? Støtter vi eller skal vi støtte språkvalg i URL-en?

Reply:
Hvis vi ikke trenger funksjonaliteten foreslår jeg å fjerne koden. Hvis vi trenger det foreslår jeg å skrive om allerede nå til noe som stemmer overens med pathingen vår:)

Reply:
>Trenger vi denne middlewaren? Støtter vi eller skal vi støtte språkvalg i URL-en?

Denne pren tar for seg både støtte i urlen og språkvelger. Mer info om internasjonalisering: https://nextjs.org/docs/app/building-your-application/routing/internationalization

>Hvis vi ikke trenger funksjonaliteten foreslår jeg å fjerne koden. Hvis vi trenger det foreslår jeg å skrive om allerede nå til noe som stemmer overens med pathingen vår:)

Bra innspill! Du kan gjerne dobbelsjekke med Kjersti. Men basert på designet og at dette monorepoet kommer til å være den nye portalen så tror jeg at vi trenger funksjonaliteten. Dette er den anbefalte måten å håndtere språk (internationalization) på i Next.js. Fra lenken over: *Routing can be internationalized by either the sub-path (/fr/products) or domain (my-site.fr/products).*

Reply:
Enig i at det høres lurt ut at vi støtter det, vi må bare bli enig om hvordan gjøre pathingen sånn at det er konsistent på tvers av løsningene.

Reply:
Jeg overlater avgjørelsen til dere. Min personlig mening er at jo nærmere vi holder oss til dokumentasjonen til rammeverket (Next.js) desto bedre er vi rustet mot teknisk gjeld og bugs, og kan bruke deres dokumentasjonsider som referanse hvis noe ikke fungerer som det skal.

Reply:
Høres ut som det var enighet om dette på møtet hvor det ble tatt opp.